scripts/gen-api-gtkdoc.py

```diff
@@ -0,0 +1,411 @@
+		d += "} %s;\n" % name
+
+		e = DoxyEnum(name, d)
+		e.add_brief(xml.find("briefdescription"))
+		for p in xml.findall("enumvalue"):
+			e.add_member(p)
+		return e
+
+class DoxyStruct(DoxyElement):
+
+	@staticmethod
+	def from_compounddef(xml, typedefs = []):
+		name = xml.find("compoundname").text
+		section = xml.find("sectiondef")
+		d = "struct %s {\n" % name;
+		for p in section.findall("memberdef"):
+			# workaround for struct members. g-ir-scanner can't properly map struct members
+			# (beginning with struct GeanyFoo) to the typedef and assigns a generic type for them
+			# thus we fix that up here and enforce usage of the typedef. These are written
+			# out first, before any struct definition, for this reason
+			# Exception: there are no typedefs for GeanyFooPrivate so skip those. Their exact
+			# type isn't needed anyway
+			s = fix_definition(p.find("definition").text).lstrip()
+			words = s.split()
+			if (words[0] == "struct"):
+				if not (words[1].endswith("Private") or words[1].endswith("Private*")):
+					s = " ".join(words[1:])
+			d += "\t%s;\n"  % s
+
+		d += "};\n"
+		e = DoxyStruct(name, d)
+		e.add_brief(xml.find("briefdescription"))
+		for p in section.findall("memberdef"):
+			e.add_member(p)
+		return e
+
+class DoxyFunction(DoxyElement):
+
+	@staticmethod
+	def from_memberdef(xml):
+		name = xml.find("name").text
+		d = normalize_text(xml.find("definition").text.replace("G_BEGIN_DECLS", ""))
+		d += " " + xml.find("argsstring").text + ";"
+		d = normalize_text(d.replace("GEANY_API_SYMBOL", ""))
+
+		e = DoxyFunction(name, d)
+		e.add_brief(xml.find("briefdescription"))
+		e.add_detail(xml.find("detaileddescription"))
+		for p in xml.xpath(".//detaileddescription/*/parameterlist[@kind='param']/parameteritem"):
+			e.add_param(p)
+		x = xml.xpath(".//detaileddescription/*/simplesect[@kind='return']")
+		if (len(x) > 0):
+			e.add_return(x[0])
+		return e
+
+other = []
+typedefs = []
+
+c_files = root.xpath(".//compounddef[@kind='file']/compoundname[substring(.,string-length(.)-1)='.c']/..")
+h_files = root.xpath(".//compounddef[@kind='file']/compoundname[substring(.,string-length(.)-1)='.h']/..")
+
+for f in h_files:
+	if not (f.find("compoundname").text.endswith("private.h")):
+		for n0 in f.xpath(".//*/memberdef[@kind='typedef' and @prot='public']"):
+			if not (n0.find("type").text.replace("G_BEGIN_DECLS", "").lstrip().startswith("enum")):
+				e = DoxyTypedef.from_memberdef(n0)
+				typedefs.append(e)
+
+	for n0 in f.xpath(".//*/memberdef[@kind='enum' and @prot='public']"):
+		e = DoxyEnum.from_memberdef(n0)
+		other.append(e)
+
+for n0 in root.xpath(".//compounddef[@kind='struct' and @prot='public']"):
+	e = DoxyStruct.from_compounddef(n0)
+	other.append(e)
+
+for f in c_files:
+	for n0 in f.xpath(".//*/memberdef[@kind='function' and @prot='public']"):
+		e = DoxyFunction.from_memberdef(n0)
+		other.append(e)
+
+outfile.write("#include <glib.h>\n")
```

Comment:
should probably have a comment stating that it is generated (same for scioutfile)